espnet2/slu/espnet_model.py

```diff
@@ -461,111 +460,20 @@ def _extract_feats(
             feats, feats_lengths = speech, speech_lengths
         return feats, feats_lengths
 
-    def nll(
+    def _calc_att_loss(
         self,
         encoder_out: torch.Tensor,
         encoder_out_lens: torch.Tensor,
         ys_pad: torch.Tensor,
         ys_pad_lens: torch.Tensor,
-    ) -> torch.Tensor:
-        """Compute negative log likelihood(nll) from transformer-decoder
-
-        Normally, this function is called in batchify_nll.
-
-        Args:
-            encoder_out: (Batch, Length, Dim)
-            encoder_out_lens: (Batch,)
-            ys_pad: (Batch, Length)
-            ys_pad_lens: (Batch,)
-        """
```

Comment:
Expect to have the comments, but you can leave it as TODO for the initial PR.

Reply:
Yeah I have added TODO!